### PR TITLE
typescript-angular: support angular 9

### PR DESCRIPTION
**Pull request by topce**
Add support for angular 9
fix api.module.ts to generate 
```typescript
ModuleWithProviders<ApiModule>
```
instead of 
```typescript
ModuleWithProviders
```
for more info check 
[migration-module-with-providers](https://angular.io/guide/migration-module-with-providers)

fixes https://github.com/OpenAPITools/openapi-generator/issues/5246

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x ] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.
@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @nicokoenig (2018/09) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02)

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAngularClientCodegen.java [hunk 1]
```diff
@@ -192,6 +193,12 @@ public void processOpts() {
             additionalProperties.put(PROVIDED_IN_ROOT, false);
         }
 
+        if (ngVersion.atLeast("9.0.0")) {                
+            additionalProperties.put(MODULE_WITH_PROVIDERS, true);
```
Comment (amakhrov) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAngularClientCodegen.java]: Looks like the purpose of this flag is to drive adding a generic param to the ModuleWithProviders (as in angular 9 the default value for param (any) has been removed). It doesn't affect whether the module with providers is generated or not. Is that correct?
If so, perhaps a better name would be "ENFORCE_GENERIC_MODULE_WITH_PROVIDERS" or smth like that?
Or maybe even remove it from the java class, and just do conditionally in the template (based on ngVersion)?

Comment (macjohnny) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAngularClientCodegen.java]: rename is fine. concerning the template condition, I think this is a bit difficult with the mustache template, so I think the current solution is fine.

**Comment by topce** [comment]
Wow one day after PR and already reviewed
And merged into master 
Simply the best!!! @macjohnny @amakhrov

**Comment by veluria** [comment]
There's a typo in TypeScriptAngularClientCodegen.java#L335 which says `8.0.0` instead of `9.0.0`.

**Comment by wing328** [comment]
@veluria do you filing a PR with the fix when you've time?

**Comment by veluria** [comment]
I can send a PR later.

**Comment by macjohnny** [comment]
@veluria thanks for pointing this out, I filed a PR to fix this: https://github.com/OpenAPITools/openapi-generator/pull/5585

**Comment by wing328** [comment]
@topce thanks for the PR, which has been included in the 4.3.0 release: https://twitter.com/oas_generator/status/1243455743937789952

**Pull request merged**